### PR TITLE
Added handle add to cart feature in category page

### PR DESCRIPTION
**Pull request by sghernandez5**
## Description

Added feature to add an item to the cart from the category page. The action is only available on Simple products. For configurable products, the button simply takes them to the PDP page, and for all other products the button is disabled.

## Related Issue

Closes PWA-1843.

### Verification Stakeholders

@revanth0212 

### Verification Steps

1. Open the deployed app.
2. Go to a simple product in the category page. Click on the add to cart button and verify the product is added to the cart.
3. Go to a configurable product. Click on the add to cart button and verify that the user is taken to the PDP page.
4. Go to any other product and verify that the button is disabled.

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by revanth0212** [comment]
@larsroettig Ignore the test failures and lint errors for now. Sofia will handle it soon.

File: packages/venia-ui/lib/components/Gallery/addToCartButton.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+const AddToCartButton = props => {
+    const talonProps = useAddToCartButton({
+        item: props.item
+    });
+    const { handleAddToCart, isDisabled, isInStock } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const buttonText = isInStock ? 'ADD TO CART' : 'OUT OF STOCK';
```
Comment (anthoula) [inline packages/venia-ui/lib/components/Gallery/addToCartButton.js:18]: For these new strings, please add translations.

**Review by anthoula — APPROVED** [review]

**Comment by fooman** [comment]
One thing to consider would be to change the Add To Cart wording on a disabled button (or not display it at all). If someone happens to get to here
https://pr-3272.pwa-venia.com/venia-accessories/venia-jewelry.html?page=1&price%5Bfilter%5D=100-200%2C100_200&fashion_material%5Bfilter%5D=Sterling+Silver%2C17
it's not immediately clear why the add to cart button would not work.

**Comment by fooman** [comment]
One other thing to consider. For the classic Luma theme when trying to add a configurable product we get the same redirect to the product page but there is a message letting the customer know that the add to cart requires more details 

> "You need to choose options for your item."

Some customer feedback would be nice here as well.

**Comment by davemacaulay** [comment]
I agree with @fooman, if you can't add it to the cart, don't show the button :)

**Comment by sghernandez5** [comment]
> I agree with @fooman, if you can't add it to the cart, don't show the button :)

@tkacheva @schensley  Kristof and Dave make great points when the button should be displayed, let me know if these changes should be made

**Comment by tkacheva** [comment]
@fooman and @sghernandez5 
we discussed the option not to show the button if product not available, but agreed that it would negatively affect PLP layout. I'm of with changing the working. @schensley?

**Comment by sghernandez5** [comment]
@pankhurigoel Hi Pankuri, developer tests are done and ready for QA.

**Review by anthoula — APPROVED** [review]

**Comment by revanth0212** [comment]
MFTF ✔️ 
Cypress ✔️

**Pull request merged**